src/sqswatcher/plugins/slurm.py

```diff
@@ -208,5 +209,50 @@ def update_cluster(max_cluster_size, cluster_user, update_events, instance_prope
     return failed, succeeded
 
 
+def _is_node_locked(hostname):
+    node_state = get_node_state(hostname)
+    log.info("Node {} currently in state {}".format(hostname, node_state))
+    for disable_state in SLURM_NODE_DISABLED_STATES:
+        if disable_state in node_state:
+            return True
+    return False
+
+
+def perform_health_actions(health_events):
+    """Update and write node lists( and gres_nodes if instance has GPU); restart relevant nodes."""
+    failed = []
+    succeeded = []
+    for event in health_events:
+        try:
+            # to-do, ignore fail to lock message if node is not in scheduler
```

Comment:
I think this is an important one. We need to make sure such actions are idempotent and do not fail if a node is not in scheduler especially since the process instance events might have just removed it.

Reply:
Right now if locking fails for any reason, we will catch it with exception, mark the event as failed, and re-queue the event. If the instance is already removed we will re-queue the event until it's eventually discarded.

Do we need to handle this case with specific logic or is using the general re-queue when failed logic ok?